planet/src/main/webapp/js/profile.js

```diff
@@ -0,0 +1,147 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+function diplaySection(sectionId) {
+    switch(sectionId) {
+        case 'about-section':
+            document.getElementById('about-section').style.display = 'block';
+            document.getElementById('posts-section').style.display = 'none';
+            document.getElementById('events-section').style.display = 'none';
+            break;
+        case 'posts-section':
+            document.getElementById('about-section').style.display = 'none';
+            document.getElementById('posts-section').style.display = 'block';
+            document.getElementById('events-section').style.display = 'none';
+            break;
+        case 'events-section':
+            document.getElementById('about-section').style.display = 'none';
+            document.getElementById('posts-section').style.display = 'none';
+            document.getElementById('events-section').style.display = 'block';
+            break;
+        default:
+            console.log('Unhandled section id:' + sectionId);
+    }
+}
+
+function activateLink(tabId) {
+    switch(tabId) {
+        case 'about-link':
+            document.getElementById('about-link').classList.add('active');
+            document.getElementById('posts-link').classList.remove('active');
+            document.getElementById('events-link').classList.remove('active');
+            break;
+        case 'posts-link':
+            document.getElementById('about-link').classList.remove('active');
+            document.getElementById('posts-link').classList.add('active');
+            document.getElementById('events-link').classList.remove('active');
+            break;
+        case 'events-link':
+            document.getElementById('about-link').classList.remove('active');
+            document.getElementById('posts-link').classList.remove('active');
+            document.getElementById('events-link').classList.add('active');
+            break;
+        default:
+            console.log('Unhandled link id:' + linkId);
+    }
+}
+
+function switchSection(linkId, sectionId) {
+    activateLink(linkId);
+    diplaySection(sectionId);
+}
+
+async function loadUserInformation(username) {
+    const usersReference = database.ref('/users');
+    const userSnapshot = await usersReference.orderByChild('username').equalTo(username).once('value');
+    if (userSnapshot.val() === null) {
+        document.getElementById('profile-page').remove();
+        document.getElementById('not-found-message').style.display = 'block';
+        return;
+    }
+    document.getElementById('not-found-message').remove();
+    document.getElementById('profile-page').style.display = 'block';
+    let user;
+    let userId;
+    for (const object in userSnapshot.val()) {
+        userId = object;
+        user = userSnapshot.val()[object];
+    }
```

Comment:
What is the point of this for loop? You just keep overwriting |userId| and |user| until you get the last one?

Reply:
The userSnapshot.val() returns an object of users. The user I want to get is only accessed through the id (which I don't have) that's why I have to loop through the properties of the object (which are user ids). Since usernames are unique, the returned object will always contain one user.

Reply:
Can you retrieve the id with something like
const userId = userSanpshot.val()[0] ? (this is where my JS knowledge reaches its limit :P)

Reply:
No! Actually that's not possible because it's an object not an array.
P.S: You can access the property "prop" of the object using: object["prop"] or object.prop, but you need the name of the property which in this case is the user id.

Reply:
Ah I see. How about rename |object| to |property| and leave a comment mentioning that userSnapShot.val() only has one property |username| is unique (also consider renaming |username| to |userId|)?